README.md

```diff
@@ -1 +1,174 @@
-# node-modules-core
\ No newline at end of file
+# Main Core Modules
+
+## Overview
+
+This lesson will cover the usage of the core Node modules. Also, we'll list the main core modules which will be useful during this course.
+
+## Objectives
+
+1. Provide a list of main core modules and their purpose
+1. Describe fs and its main methods
+1. Describe url and its main methods
+1. Describe path and path.join
+
+## Core Modules
+
+As discussed previously, Node is a platform or an environment in which we build applications. You saw that modularized code is better and how you can write your own modules. With modules, developers from different team and different companies can share their code (what is called open-source community). This is great but how do you get the code? Imagine a simple thing like parsing a path, i.e., we have a string like `/home/user/azat/program.js` and we need to extract the extension, base, file name, etc.
+
+You would have to write a code which uses Regular Expressions like this:
+
+```js
+// Split a filename into [root, dir, basename, ext], unix version
+// 'root' is just a slash, or nothing.
+const splitPathRe =
+    /^(\/?|)([\s\S]*?)((?:\.{1,2}|[^\/]+?|)(\.[^.\/]*|))(?:[\/]*)$/;
+var posix = {}
+
+function posixSplitPath(filename) {
+  const out = splitPathRe.exec(filename);
+  out.shift()
+  return out
+}
+
+posix.parse = function(pathString) {
+  var allParts = posixSplitPath(pathString);
+  return {
+    root: allParts[0],
+    dir: allParts[0] + allParts[1].slice(0, -1),
+    base: allParts[2],
+    ext: allParts[3],
+    name: allParts[2].slice(0, allParts[2].length - allParts[3].length)
+  }
+}
+```
+
+This code is only for Unix-like systems (Linux, Mac OS X, etc.) and for Windows you would have to write a different parser. Do you realize that every time you want to access a file you would need something like this? Probably you won't copy/paste the code, but have a module. If you need this module so often maybe it's a good idea to make it part of the system? That's why creators of Node, packaged this path parser and other utilities and libraries that are likely to be used in most Node projects together into the platform itself. These packaged modules called core modules.
+
+Even more so, more complex core modules like `net` depend on a simpler core modules like `path`. 
```

Comment:
Add header here called "Core Modules" - add transition sentence, something like "So how do we use this `path` modue?